dapr_agents/observability/instrumentor.py

```diff
@@ -262,12 +263,28 @@ async def context_wrapped_coro():
                     return loop.run_until_complete(context_wrapped_coro())
                 except RuntimeError:
                     # No running loop - create new one
-                    loop = asyncio.new_event_loop()
-                    asyncio.set_event_loop(loop)
                     try:
+                        loop = asyncio.new_event_loop()
+                        asyncio.set_event_loop(loop)
                         return loop.run_until_complete(context_wrapped_coro())
+                    except Exception as e:
+                        logger.warning(
```

Comment:
this section, and maybe the whole context_wrap needs to be analyze in the context of where is this executed in workflows and/or activities. it is strange for new_event_loop to fail, it might indicate other subtle bug,

Reply:
This part of the code is managing OpenTelemetry context across thread boundaries. I'll add a TODO comment on this, but I think when we get to the point of using trace context propagation through Dapr all of this will be resolved